tidy3d/plugins/design/method.py

```diff
@@ -180,23 +124,430 @@ def sample(self, parameters: Tuple[ParameterType, ...], **kwargs) -> Dict[str, A
 
         # sample each dimension individually
         vals_each_dim = {}
-        for design_var in parameters:
-            vals = design_var.sample_grid()
-            vals_each_dim[design_var.name] = vals
+        for param in parameters:
+            vals = param.sample_grid()
+            vals_each_dim[param.name] = vals
 
         # meshgrid each dimension's results and combine them all
         vals_grid = np.meshgrid(*vals_each_dim.values())
         vals_grid = (np.ravel(x).tolist() for x in vals_grid)
-        return dict(zip(vals_each_dim.keys(), vals_grid))
+        vals_dict = dict(zip(vals_each_dim.keys(), vals_grid))
+        t_vals_dict = [dict(zip(vals_dict.keys(), values)) for values in zip(*vals_dict.values())]
+
+        return t_vals_dict
+
+
+class MethodOptimise(Method, ABC):
+    """A method for handling design searches that optimise the design"""
+
+    # NOTE: We could move this to the Method base class but it's not relevant to MethodGrid
+    rng_seed: pd.PositiveInt = pd.Field(
+        title="Seed for random number generation", description="TBD", default=None
+    )
+
+    def any_to_int_param(self, parameter):
+        """Convert ParameterAny object to integers and provide a conversion dict to return"""
+
+        return dict(enumerate(parameter.allowed_values))
+
+    def sol_array_to_dict(
+        self, solution: np.array, keys: list, param_converter: dict
+    ) -> list[dict]:
+        """Convert an array of solutions to a list of dicts for function input"""
+        sol_dict_list = [dict(zip(keys, sol)) for sol in solution]
+
+        self._handle_param_convert(param_converter, sol_dict_list)
+
+        return sol_dict_list
+
+    def _handle_param_convert(self, param_converter, sol_dict_list):
+        for param, convert in param_converter.items():
+            for sol in sol_dict_list:
+                if isinstance(sol[param], float):
+                    sol[param] = int(round(sol[param], 0))
+                sol[param] = convert[sol[param]]
 
 
-class AbstractMethodRandom(MethodIndependent, ABC):
+class MethodBayOpt(MethodOptimise, ABC):
+    """A standard method for performing bayesian optimization search"""
+
+    initial_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of initial random search iterations.",
+        description="TBD",
+    )
+
+    n_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of bayesian optimization iterations.",
+        description="TBD",
+    )
+
+    acq_func: Optional[Literal["ucb", "ei", "poi"]] = pd.Field(
+        title="Type of acquisition function.",
+        description="TBD",
+        default="ucb",
+    )
+
+    kappa: Optional[pd.PositiveFloat] = pd.Field(
+        title="Kappa parameter for the Gaussian processor.",
```

Comment:
for optional, we should set a default value, even if it's just `None`

Reply:
It does have a default, I've just included it as a keyword argument below.

Reply:
ah I see! that's maybe a better way to do things to be honest. but since we generally put the default value as first argument in the fields, maybe can you move this kwarg to be the first one so it's more clear internally?